### PR TITLE
fix(badge): DP-102533 mention only color

### PR DESCRIPTION
**Pull request by danieldialpad**
# PR Title

## Obligatory GIF (super important!)

Obligatory GIF

## :hammer_and_wrench: Type Of Change

These types will increment the version number on release:

- [x] Fix
- [ ] Feature
- [ ] Performance Improvement
- [ ] Refactor

These types will not increment the version number, but will still deploy to documentation site on release:

- [ ] Documentation
- [ ] Build system
- [ ] CI
- [ ] Style (code style changes, not css changes)
- [ ] Test
- [ ] Other (chore)

## :book: Jira Ticket

https://dialpad.atlassian.net/browse/DP-102533

## :bulb: Context

This is a change is necessary for a corner case missed in the design. 
This corner case is:
The user has the channel setting set as 'All Messages'. There is a situation where the user only receives  MENTION messages. It was assumed that the color should be #7C52FF. As pointed out by QA and after discussing with the designer, in this situation, the color should be #3A1D95.

Since we are not changing the current behavior of badge count when the channel setting is Mention only or Muted, I had to add a new prop 'channelSetting' to count for this situation. When the channel setting is 'All message' and there is unread mention count and NO unread count, the color will set to be #3A1D95

## :pencil: Checklist

For all PRs:

- [ ] I have ensured no private Dialpad links or info are in the code or pull request description (Dialtone is a public repo!).
- [ ] I have reviewed my changes.
- [ ] I have added all relevant documentation.
- [ ] I have considered the performance impact of my change.

For all Vue changes:

- [ ] I have added / updated unit tests.
- [x] I have made my changes in Vue 2 and Vue 3. Note: you may sync your changes from Vue 2 to Vue 3 (or vice versa) using the `./scripts/dialtone-vue-sync.sh` script. Read docs here: [Dialtone Vue Sync Script](../packages/dialtone-vue3/.github/CONTRIBUTING.md#dialtone-vue-sync-script)
- [x] I have validated components with a screen reader.
- [ ] I have validated components keyboard navigation.

For all CSS changes:

- [x] I have used design tokens whenever possible.
- [ ] I have considered how this change will behave on different screen sizes.
- [x] I have visually validated my change in light and dark mode.
- [ ] I have used gap or flexbox properties for layout instead of margin whenever possible.

If new component:

- I am exporting any new components or constants:
  - [ ] from the index.js in the component directory.
  - [ ] from the index.js in the root (either `packages/dialtone-vue2` or `packages/dialtone-vue3`).
- [ ] I have added the styles for the new component to the `packages/dialtone-css` package.
- [ ] I have created a page for the new component on the documentation site in `apps/dialtone-documentation`.
- [ ] I have added the new component to `common/components_list.cjs`
- [ ] I have created a component story in storybook
- [ ] I have created story / stories for any relevant component variants in storybook
- [ ] I have created a docs page for the component in storybook.
- [ ] I have checked that changing all props/slots via the UI in storybook works as expected.

### CODE REVIEW

File: packages/dialtone-vue3/recipes/leftbar/general_row/general_row.vue
Comment (ninamarina) [inline packages/dialtone-vue3/recipes/leftbar/general_row/general_row.vue:220]: Add in the comment the posible values since we fill the automated docs with this:
```suggestion
     *  @values 'mention', 'always', null
     */
```
If null is not an option, then it should also be marked as `required`

**Comment by ninamarina** [comment]
Add the Jira ticket to the title, instead of `NO-JIRA`, this should be `DP-102533`

**Comment by francisrupert** [comment]
> Add the Jira ticket to the title, instead of `NO-JIRA`, this should be `DP-102533`

Title should also include the scope, i.e. `fix(badge): DP-102533 mention only color`

**Comment by iropolo** [comment]
Seems like this is the update.
unreadMentionCount: 5
hasUnreads: true 
channelSetting: 'always'

image

If it get design approval should be ok to go. Ping @fede-dp

**Review by iropolo — APPROVED** [review]
Wait for design approval.

**Review by francisrupert — APPROVED** [review]
Design is approved. Defer to Nina's comments otherwise. Thanks!

**Review by braddialpad — APPROVED** [review]

**Pull request merged**
